drivers/counter/counter_mspm0gxxxx.c

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) 2024 Bang & Olufsen A/S, Denmark
+//
+// SPDX-License-Identifier: LicenseRef-BnOProprietary
+
+#include <ti/driverlib/dl_timerg.h>
+#include <ti/driverlib/dl_timer.h>
+#include <zephyr/drivers/counter.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/counter.h>
+
+LOG_MODULE_REGISTER(counter_mspm0gxxxx, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define DT_DRV_COMPAT ti_mspm0gxxxx_counter
+// NOTE: it seems(?) that the clocks support multiple input channels but
+// they must be configured on capture mode
```

Comment:
Potato potato maybe :)
```suggestion
// they must be configured with capture mode
```